File: Backend/Iteramico/DomainRepository/Models/JourneyParticipation.cs
using System.Text.Json.Serialization;

namespace DomainRepository.Models
{
#nullable disable
    public class JourneyParticipation
    {
        public DateTime JoinedOn { get; set; }

        public int UserId { get; set; }
        [JsonIgnore]
        public virtual User User { get; set; }
        public int JourneyId { get; set; }
        public virtual Journey Journey { get; set; }
    }
}

File: frontend/lib/models/journey_participation.dart
import 'package:frontend/models/journey.dart';

class JourneyParticipation {
  final DateTime joinedOn;
  final int userId;
  final Journey journey;

  JourneyParticipation(
      {required this.joinedOn, required this.userId, required this.journey});

  factory JourneyParticipation.fromJson(Map<String, dynamic> json) {
    return JourneyParticipation(
      joinedOn: DateTime.parse(json['joinedOn']),
      userId: json['userId'],
      journey: Journey.fromJson(json['journey']),
    );
  }
}
